packages/synapse-react-client/src/utils/hooks/useDetectSSOCode.test.ts

```diff
@@ -384,4 +387,47 @@ describe('useDetectSSOCode tests', () => {
       consoleSpy.mockReset()
     },
   )
+
+  test('Does not attempt to use the state param if we are acting as identity provider', async () => {
```

Comment:
This is kind of a weird test, but it checks that we maintain the existing behavior.

In this scenario, we are both acting as an IdP and simultaneously using an external IdP to sign into a Synapse account. In this case we do not do anything with the `state` parameter. 

The 'proper' behavior would probably be that we capture and serialize client's `state` data in our own `state` object that we send to the external IdP (along with any other data we might want to put in `state`), then deserialize the data and send it back to the client, but that's a more complex change than necessary to fix the regression.

I captured that we should test this more thoroughly in https://sagebionetworks.jira.com/browse/PORTALS-2999?focusedCommentId=207029